b2g/components/RemoteControlService.jsm

```diff
@@ -0,0 +1,395 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+ * RemoteControlService.jsm is the entry point of remote control function.
+ * The service initializes a TLS socket server to receives events from user.
+ *
+ *               RemoteControlService <-- Gecko Preference
+ *
+ *     user -->  nsITLSSocketServer --> script (gecko)
+ *
+ * Events from user are in JSON format. After parsed to control command,
+ * these events are passed to script (js), runs in sandbox,
+ * and dispatch corresponding events to Gecko.
+ *
+ * Here is related component location:
+ * gecko/b2g/components/RemoteControlService.jsm
+ *
+ * For more details, please visit: https://wiki.mozilla.org/Firefox_OS/Remote_Control
+ */
+
+"use strict";
+
+this.EXPORTED_SYMBOLS = ["RemoteControlService"];
+
+const { classes: Cc, interfaces: Ci, results: Cr, utils: Cu, Constructor: CC } = Components;
+
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/debug.js");
+
+XPCOMUtils.defineLazyServiceGetter(this, "certService", "@mozilla.org/security/local-cert-service;1",
+                                   "nsILocalCertService");
+
+// static functions
+function debug(aStr) {
+  dump("RemoteControlService: " + aStr + "\n");
+}
+
+const DEBUG = false;
+
+const REMOTE_CONTROL_EVENT = 'mozChromeRemoteControlEvent';
+const RC_PREF_DEVICES = "remotecontrol.authorized_devices";
+const MAX_CLIENT_CONNECTIONS = 5; // Allow max 5 clients to use remote control TV
+
+const SERVER_STATUS = {
+  STOPPED: 0,
+  STARTED: 1
+};
+
+this.RemoteControlService = {
+  // Remote Control status
+  _serverStatus: SERVER_STATUS.STOPPED,
+
+  // TLS socket server
+  _port: -1, // The port on which this service listens
+  _socket: null, // The socket associated with this
+  _doQuit: false, // Indicates when the service is to be shut down at the end of the request.
+  _socketClosed: true, // True if the socket in this is closed, false otherwise.
```

Comment:
`_socketClosed` isn't really used in your code, you can simply remove this attribute.